packages/core/src/components/graphs/radar.ts

```diff
@@ -0,0 +1,473 @@
+// Internal Imports
+import { Component } from "../component";
+import { DOMUtils } from "../../services";
+import * as Configuration from "../../configuration";
+import { Events, TooltipTypes, Roles } from "../../interfaces";
+import { Tools } from "../../tools";
+import { flatMapDeep, kebabCase } from "lodash-es";
+import {
+	Point,
+	Angle,
+	radialLabelPlacement,
+	radToDeg,
+	polarToCartesianCoords,
+	distanceBetweenPointOnCircAndVerticalDiameter
+} from "../../services/angle-utils";
+
+// D3 Imports
+import { select } from "d3-selection";
+import { scaleBand, scaleLinear, ScaleLinear } from "d3-scale";
+import { max, extent } from "d3-array";
+import { lineRadial, curveLinearClosed } from "d3-shape";
+
+// used to make transitions
+let oldYScale: ScaleLinear<number, number>;
+
+interface Datum {
+	group?: string;
+	key: string;
+	value: number;
+}
+interface GroupedDatum {
+	name: string;
+	data: Array<Datum>;
+}
+
+export class Radar extends Component {
+	type = "radar";
+	svg: SVGElement;
+	groupMapsTo: string;
+	uniqKeys: string[];
+	uniqGroups: string[];
+	displayDataNormalized: Array<Datum>;
+	groupedDataNormalized: Array<GroupedDatum>;
+
+	init() {
+		const { events } = this.services;
+		// Highlight correct line legend item hovers
+		events.addEventListener(Events.Legend.ITEM_HOVER, this.handleLegendOnHover);
+		// Un-highlight lines on legend item mouseouts
+		events.addEventListener(Events.Legend.ITEM_MOUSEOUT, this.handleLegendMouseOut);
+	}
+
+	render(animate = true) {
+		const self = this;
+
+		this.svg = this.getContainerSVG();
+		const { width, height } = DOMUtils.getSVGElementSize(this.parent, { useAttrs: true });
+		if (!width || !height) {
+			return;
+		}
+
+		const data: Array<Datum> = this.model.getData();
+		const displayData: Array<Datum> = this.model.getDisplayData();
+		const groupedData = this.model.getGroupedData();
```

Comment:
why do we need `data`, `displayData`, `groupedData`, `displayDataNormalized` and `groupedDataNormalized`?

Reply:
We need `data` to compute the `uniqueKeys` which must always be the same, they must not change if datasets change (so we can't use `displayData`).
`displayData` is used to show only the groups that the user chooses to see by filtering them using the legend.
`groupedData` is used to compute `groupedDataNormalized`.
`displayDataNormalized` and `groupedDataNormalized` are `displayData` and `groupedData` respectively, in which missing data is added with `value = null`. Doing so we can show always the blob, even if some data are missing.

Reply:
seems like `groupedDataNormalized` & `displayDataNormalized` should move into `model.ts` 🙂

Reply:
Ideally called `getGroupedData` can take in either `{ normalize: true }` config or we need a `getNormalizedGroupedData()`

Reply:
We tried to make this more generic, but we obtained a function with 4 parameters. Being that it's maybe useless for other charts, we would not refactor it into `model.ts` to avoid creating unnecessary complexity